### ISSUE TITLE
Internal listener port should not affect dereferenced resources

### ISSUE THREAD

**Issue by retog** [zazukoians/trifid-ld#9]
I just changed a config to listen to port 8080 instead of 80. i redirected the port so that externally it was still visible under port 80, nevertheless the resource trifid tried to dereference was http://hostname:8080/path rather than the actually accessed http://hostname/path. I think trifid should not try to be smart and in the absence of specific rewriting config assumed the requested URI should be used.

**Comment by ktk** [comment]
That doesn't really make sense, where would it get the 8080 from? Nevertheless we have to fix the whole proxy/forwarding stuff once so it will work in all environments out of the box. This is not the case yet right now.

**Comment by retog** [comment]
```
That doesn't really make sense, where would it get the 8080 from?
```

Obviously from the listener/port config.

**Comment by ktk** [comment]
Ah now I get it, it's forwarded in Docker. We need to see what the request header looks like in Trifid in that case, not sure if it even gets the original request URI somewhere.

**Comment by bergos** [comment]
I don't know your configuration, but I expect you just define the port on the listener level and not on the application/root level, right? Please have a look at the example [config.js](https://github.com/zazukoians/trifid-ld/blob/master/config.js#L95). Bulding a full URL from an HTTP request requires always to be smart, because of security reasons. The proxy configuration should be mentioned in the readme...

**Comment by retog** [comment]
@bergos clearly trifid needs to know where to listen to, no question on this. But it should reconstruct the requested URI exclusively from the HTTP-Request, i.e. from the URI-Path, the host header and by the x-forwarded-proto header (or its absence). One trifid-ld instance should be allowed to provide dereferenciation for http://example.org/foo, http://example.org:7534:/bar and https://example.com/

I think the configuration should be kept simpler, if one has security concerns and wants to prevent user from getting the description of resource with the "wrong" hostname one should add this checks on a proxy level. So the configuration options you link to should imho be removed.

**Comment by bergos** [comment]
Will be implemented in v1: [#31](https://github.com/zazukoians/trifid-ld/issues/31)
